node/td-utils/test/examples/httpAndMqtt.json

```diff
@@ -0,0 +1,33 @@
+{
+    "id": "urn:simple",
+    "@context": "https://www.w3.org/2022/wot/td/v1.1",
+    "title": "MyLampThing",
+    "description": "Valid TD copied from the spec's first example",
+    "securityDefinitions": {
+        "basic_sc": { "scheme": "basic", "in": "header" }
+    },
+    "security": ["basic_sc"],
+    "properties": {
+        "status": {
+            "type": "string",
+            "forms": [{ "href": "http://mylamp.example.com:1234/status" }]
+        }
+    },
+    "actions": {
+        "toggle": {
+            "forms": [{ "href": "mqtt://mylamp.example.com:1234/toggle" }]
+        }
+    },
+    "protocolSchemes": [
+        {
+            "protocol": "http",
+            "hostname": "mylamp.example.com",
+            "port": 1234
+        },
+        {
+            "protocol": "mqtt",
+            "hostname": "mylamp.example.com",
+            "port": 1234
+        }
+    ]
```

Comment:
On the one hand I can see the nice effect of having TD and the expected test results _close_ to each other on the other had I am not sure about it.

My personal opinion is more to have the check like the following in the test.

```
[
        {
            "protocol": "http",
            "hostname": "mylamp.example.com",
            "port": 1234
        },
        {
            "protocol": "mqtt",
            "hostname": "mylamp.example.com",
            "port": 1234
        }
    ]
```

Anyway, no strong opinion. @egekorkan ?

Reply:
Last week, we checked json-schema's [test suite](https://github.com/json-schema-org/JSON-Schema-Test-Suite/blob/main/tests/draft7/additionalItems.json) with @egekorkan. I think our plan is to have similar test suites in the future that separate the TD, TM, etc., under test from the expected result of the test. Personally, I don't like putting the expected behavior of a test inside a TD since the TD becomes something else afterward. However, I like having them close to each other because I think it is easier for the reader to understand how we get the expected result. I implemented [here](https://github.com/eclipse-thingweb/td-tools/pull/32/commits/a19787d9399eabe27e07fdcb6e3244d57fa783f1) something similar what they did in json-schema test suite. With this implementation, TDs under examples are pure without any additional fields, and the test suite is separated into a .ts file.